src/main/java/org/apache/commons/compress/compressors/zstandard/ZstdUtils.java

```diff
@@ -28,7 +31,31 @@
  */
 public class ZstdUtils {
 
-    enum CachedAvailability {
+    private static final class SizeMeasurementOutputStream extends OutputStream {
+        long size = 0;
+        private OutputStream out;
+
+        public SizeMeasurementOutputStream() { }
+
+        /**
+         * Can be null, if the output data should be counted. The close of this stream wont close its delegated output stream.
+         * 
+         * @param delegate the output stream to which the output will be delegated
+         */
+        public SizeMeasurementOutputStream(OutputStream delegate) {
```

Comment:
If the input is null, then don't construct this instance in the first place, and you won't pay the price of an extra level of indirection. 

This kind of stream should really be a kind of `java.io.FilterOutputStream` like `org.apache.commons.io.output.CountingOutputStream` (where you'd override the close method).